src/processor/operator/index_lookup.cpp

```diff
@@ -0,0 +1,108 @@
+#include "processor/operator/index_lookup.h"
+
+#include "storage/copier/table_copy_utils.h"
+#include "storage/index/hash_index.h"
+#include <arrow/array/array_base.h>
+#include <arrow/type.h>
+
+using namespace kuzu::common;
+using namespace kuzu::storage;
+
+namespace kuzu {
+namespace processor {
+
+bool IndexLookup::getNextTuplesInternal(ExecutionContext* context) {
+    if (!children[0]->getNextTuple(context)) {
+        return false;
+    }
+    for (auto& info : infos) {
+        assert(info);
```

Comment:
info is a reference, i think we can remove the assert

Reply:
This is to assert the unique_ptr because we are dereferencing its value next in L20.